src/Items/ItemBoat.h

```diff
@@ -94,8 +94,43 @@ class cItemBoatHandler :
 			return false;
 		}
 
+		// Get the boat type
+		cBoat::eBoatType BoatType = cBoat::btOak;
+		switch (a_Player->GetEquippedItem().m_ItemType)
+		{
+			case E_ITEM_SPRUCE_BOAT:
+			{
+				BoatType = cBoat::btSpruce;
+				break;
+			}
+			case E_ITEM_BIRCH_BOAT:
+			{
+				BoatType = cBoat::btBirch;
+				break;
+			}
+			case E_ITEM_JUNGLE_BOAT:
+			{
+				BoatType = cBoat::btJungle;
+				break;
+			}
+			case E_ITEM_ACACIA_BOAT:
+			{
+				BoatType = cBoat::btAcacia;
+				break;
+			}
+			case E_ITEM_DARK_OAK_BOAT:
+			{
+				BoatType = cBoat::btDarkOak;
+				break;
+			}
+			default:
+			{
+				break;
+			}
+		}
```

Comment:
This switch might be useful as a separate static method in the `cBoat` class, possibly exported to Lua.

Now it occurred to me, do we have boat spawning by a dispenser implemented? That should make use of this, too.